docs/pages/guides/onchain-indices.mdx

```diff
@@ -0,0 +1,412 @@
+    require(positionIsEmpty, "another player is at this position");
+
+    Position.set(player, x, y);
+  }
+}
+```
+
+   </CollapseCode>
+
+### Using modules
+
+#### `KeysWithValue`
+
+We can also use a [Module](/world/modules) to install and manage an index for us. The [`KeysWithValue`](/world/modules/keyswithvalue) module can be installed on tables and maintains an array of keys with each value.
+
+In this case, we install it for the `Position` table, in the `modules` property of the MUD config:
+
+   <CollapseCode>
+```ts filename="mud.config.ts" copy showLineNumbers {12-18}
+import { mudConfig } from "@latticexyz/world/register";
+
+export default mudConfig({
+	tables: {
+		Position: {
+		  keySchema: {
+		    player: "address",
+		  },
+		  valueSchema: { x: "int32", y: "int32" },
+		}
+	},
+  modules: [
+    {
+      name: "KeysWithValueModule",
+      root: true,
+      args: [resolveTableId("Position")],
+    },
+  ],
+}
+```
+
+  </CollapseCode>
+
+Internally, this module uses [Store Hooks](https://mud.dev/store/store-hooks) to automatically update the index whenever a `Position` record is set.
+
+After the module is installed, we can use a helper to get all addresses at a given position:
+
+```solidity filename="MoveSystem.sol" copy showLineNumbers {4,6-7,13-18}
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import { PackedCounter } from "@latticexyz/store/src/PackedCounter.sol";
+import { System } from "@latticexyz/world/src/System.sol";
+import { getKeysWithValue } from "@latticexyz/world-modules/src/modules/keyswithvalue/getKeysWithValue.sol";
+import { Position, PositionTableId } from "../codegen/index.sol";
+
+contract MoveSystem is System {
+  function move(int32 x, int32 y) public {
+    address player = _msgSender();
+
+    bool positionIsEmpty = getKeysWithValue(
+      PositionTableId,
+      Position.encodeStatic(x, y),
+      PackedCounter.wrap(bytes32(0)),
+      ""
+    ).length == 0;
+    require(positionIsEmpty, "another player is at this position");
+
+    Position.set(player, x, y);
+  }
+}
+```
+
+### Gas comparison
+
+With MUD's [`GasReporter`](https://github.com/latticexyz/mud/tree/main/packages/gas-report), we can determine the gas use of `MoveSystem.move`:
+
+```solidity filename="MoveSystem.t.sol" copy showLineNumbers
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import { MudTest } from "@latticexyz/world/test/MudTest.t.sol";
+import { GasReporter } from "@latticexyz/gas-report/src/GasReporter.sol";
+
+import { IWorld } from "../src/codegen/world/IWorld.sol";
+
+contract MoveSystemTest is MudTest, GasReporter {
+  function testMove() public {
+    startGasReport("moving one player");
+    IWorld(worldAddress).move(1, 1);
+    endGasReport();
+  }
+}
+```
+
+Comparing the three approaches, we find that managing our own indices is generally more gas-efficient, with the "position is occupied" index being the cheapest by far.
+
+#### `PlayersAtPosition`
```

Comment:
If you decide to delete `PlayersAtPosition`, delete this one too. Otherwise, change as requested in the tables below.